### PR TITLE
Fix issue #297 adding support for constant visibility into the parser for PHP 7.1

### PR DESCRIPTION
**Pull request by mmucklo**
This should add support for constant visibility in PHP 7.1.

### CODE REVIEW

File: src/test/resources/files/Source/Language/PHP/PHPParserVersion71/testConstVisibility.php [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<?php
+
+class A {
+    public const ABCD = 1;
+    protected const CDEF = 2;
+    private const ABC_DEF = "some string";
+}
```
Comment (roukmoute) [inline src/test/resources/files/Source/Language/PHP/PHPParserVersion71/testConstVisibility.php]: Missing carriage return.

Comment (mmucklo) [inline src/test/resources/files/Source/Language/PHP/PHPParserVersion71/testConstVisibility.php]: ok, will fix

**Comment by mmucklo** [comment]
Ok, carriage return fixed, anything else?

File: src/main/php/PDepend/Source/Language/PHP/PHPParserVersion71.php [hunk 2]
```diff
@@ -125,4 +126,22 @@ protected function parseTypeHint()
...
+    /**
+     * Override this in later PHPParserVersions as necessary
+     * @param integer $tokenType
+     * @param integer $modifiers
+     * @return \PDepend\Source\AST\ASTConstantDefinition;
+     * @throws UnexpectedTokenException
+     */
+    protected function parseUnknownDeclaration($tokenType, $modifiers)
+    {
+        switch ($tokenType) {
```
Comment (babaorum) [inline src/main/php/PDepend/Source/Language/PHP/PHPParserVersion71.php]: Maybe change this switch with only one case to an if condition.

If you really want to keep a switch for further conditions, please call `parent::parseUnknownDeclaration($tokenType, $modifiers)` as the default case.

File: src/test/resources/files/Source/Language/PHP/PHPParserVersion71/testConstVisibility.php (hunk 1 above)
Comment (mmucklo) [inline src/test/resources/files/Source/Language/PHP/PHPParserVersion71/testConstVisibility.php]: fixed

File: src/main/php/PDepend/Source/Language/PHP/PHPParserVersion71.php (hunk 2 above)
Comment (mmucklo) [inline src/main/php/PDepend/Source/Language/PHP/PHPParserVersion71.php]: Should be addressed.

**Comment by mmucklo** [comment]
Both issues raised were addressed - anything else?

**Comment by mmucklo** [comment]
Any update?

**Comment by babaorum** [comment]
Travis seem to have some issues with your changes. Maybe that is why there is no update for now.

**Comment by mmucklo** [comment]
Sigh,

Travis would fail the master branch if run again right now.

If you take a closer look, you'd see the issues Travis has with the build are:

1) hhvm build

The hhvm build fails because pdepend no longer supports hhvm and
PR#309 has not yet been merged (addressing issue #308).

Alternatively I could open a separate PR to fix the hhvm build
properly, but fixing it in this PR is out of scope as it's broken in
master right now.

2) Travis no longer supports php 5.3 (so the 5.3 build fails).

I can remove 5.3 support for pdepend in this PR, but it's kind of out
of the scope of this change as well.

@baborum, what do you suggest I do?

**Comment by babaorum** [comment]
My bad @mmucklo , I saw that php 5.3 was dropped, but not that HHVM was already failing.

I don't know any way to speed things up here ... sorry.
The last commit seems to be from 7 months ago ...

All you could do is find a way to contact @manuelpichler for this.

**Comment by niconoe-** [comment]
A chance that last conversation is 11 days ago while I need this bugfix... I thought this will be never fixed and I don't know how I would fix it. Actually, the changes you made are awesome compared to what I tried on my local machine...

@manuelpichler is it possible to accept quickly the PR#309 as it is only deletion of HHVM so very easy to review and accept?
Then, it would be great to accept this PR (#314) even if the Travis fails as already explained by @mmucklo , the PHP 5.3 support is over for Travis right now?

Thank you.

**Comment by mmucklo** [comment]
You can importing my fork following instructions similar to:

http://tech.osteel.me/posts/2015/08/10/how-to-use-the-fork-of-a-repository-with-composer.html

But if you're using PHPMD, you'd have to use "pdepend/pdepend": "dev-master as 2.5" or something like that in order to for composer to import it properly without complaint about conflicting versions.

**Comment by niconoe-** [comment]
That is nice, I'll try to.
But the pdepend tool I use is loaded inside a docker-container downloading the last version of pdepend. I can write a Dockerfile that will install composer, download your fork and uninstall composer (to light the image) but it is very heavy working process while the maintainer just have to click twice "accept PR"...
If I'll be really in need, I'll do this for sure ;).
Thx again

**Comment by omissis** [comment]
hey guys is there anything we can do to help with moving this PR forward? :)

**Comment by emirb** [comment]
Thanks @mmucklo.

**Pull request merged**